## Depth blob detection parameters (all are also dynamically reconfigurable)
# Parameters are listed in the order in which the corresponding
# operations are executed (except for deprecated parameters).

# image preprocessing
roi/x_offset: 0
roi/y_offset: 0
roi/width: 0
roi/height: 0

unknown_pixel_value: 65535
structuring_element_a: 5
structuring_element_b: 3
dilate_iterations: 2
erode_iterations: 10
erode_ignore_empty_iterations: 0

gaussianblur_size: 0
medianblur_size: 0

# filtration of detected blobs
filter_by_area: true
min_area: 500
max_area: 10000

filter_by_circularity: false
min_circularity: 0.3
max_circularity: 0.9

filter_by_orientation: true
min_angle: 0.0
max_angle: 0.4

filter_by_inertia: false
min_inertia_ratio: 0.2
max_inertia_ratio: 0.8

filter_by_convexity: false
min_convexity: 0.75
max_convexity: 1.0

filter_by_color: true
min_depth: 1000
max_depth: 50000

filter_by_known_pixels: true
min_known_pixels_ratio: 0.5

# further filtration
min_dist_between: 10.0
max_area_diff: 10000
min_repeatability: 4

threshold_step: 1000

# deprecated parameters (do not work very well/replaced by better methods)
use_threshold_width: false
threshold_width: 100
blur_empty_areas: false
